### ISSUE TITLE
Create test set

### ISSUE THREAD

**Issue by rafafrdz** [rafafrdz/criteria4s#11]

**Comment by mark-kar** [comment]
Can I share a few thoughts about this issue?

I think we can try to add some tests to the core of the library (but actually I do not imagine what it should be looks like😄).
And I think we can add testcontainers here for checking the dialects. There we can define different criterias, interpret it by all the dialects and check if it is correct by launching the instance of db (postgres, mongo and so on). What do you think? 

In your opinion, which tests the project needs?

**Comment by rafafrdz** [comment]
You're right! My idea was once we get the proper grammar for some dialect, we would give some unit test that matchs the criteria result and the expected expression. That is just to be aware of future possible modifications.

On the other hand, what you say about testcontainers looks a great idea. For my it is okay and go ahead! However, it would be tricky (maybe) for some non-easy accessible datastores, for example private datastores as Microsoft SQL Server, Apache Impala or Apache Hive (?)

Nevertheless, if we can cover most of them, using testcontainer is a good way 😃

**Comment by mark-kar** [comment]
Cool! I'll take this issue then :)

**Comment by rafafrdz** [comment]
BTW, I created a discussion just for maintainers to organized ideas or anything about the library (or whatever) instead of chat each other in issues or PRs.

Let me know if you have access: https://github.com/rafafrdz/criteria4s/discussions/31
